dts/arm/cypress/psoc6.dtsi

```diff
@@ -0,0 +1,83 @@
+#include <cypress/mem.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m0+";
+			reg = <0>;
+		};
+		cpu@1 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m4f";
+			reg = <1>;
+		};
+	};
+
+	flash-controller@4001E000 {
+			compatible = "nrf,nrf51-flash-controller";
+			reg = <0x4001E000 0x518>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			label="NRF_FLASH_DRV_NAME";
+
+			flash0: flash@10000000 {
+				compatible = "soc-nv-flash";
+				label = "NRF_FLASH";
+				reg = <0x10000000 DT_FLASH_M0P_SIZE>;
+				write-block-size = <4>;
+			};
+
+			flash1: flash@10060000 {
+				compatible = "soc-nv-flash";
+				label = "NRF_FLASH";
+				reg = <0x10060000 DT_FLASH_M4_SIZE>;
+				write-block-size = <4>;
+			};
+	};
+
+	sram0: memory@8000000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0x08000000 DT_SRAM_M0P_SIZE>;
+	};
+
+	sram1: memory@8023000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0x08023000 DT_SRAM_SHAR_SIZE>;
+	};
+
+	sram2: memory@8024000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0x08024000 DT_SRAM_M4_SIZE>;
+	};
+
+	soc {
+		uart5: uart@40001000 {
+			compatible = "cypress,nrf-uart";
```

Comment:
The name here should be "cypress,psoc6-uart". (Unless you know that it's the same UART as in psoc5, psoc4, etc., in which case it can be "cypress,psoc-uart".)

Reply:
Already fixed in "Refactored Cypress PSoC6 UART driver" commit

Reply:
> Already fixed in "Refactored Cypress PSoC6 UART driver" commit

Yeah, I figured that later too. That's why, as said in https://github.com/zephyrproject-rtos/zephyr/pull/8408#issuecomment-397592114 and https://github.com/zephyrproject-rtos/zephyr/pull/8408#issuecomment-414376131 , it's important to squash fixup into the main commit, so people looking at a particular commit, so the latest, clean code.

Because otherwise, just consider yourself: this PR changes/adds 304 files. It's very hard to review it all at once. Even knowing that there's e.g. vendor HAL, which should be skipped, it's hard to find files which should be reviewed, and which not.

The only realistic way is to review commit by commit, and then it's important that fix-ups are squashed into each separate commits they affect.